lib/usart/src/stm32h7.rs

```diff
@@ -0,0 +1,149 @@
+// This Source Code Form is subject to the terms of the Mozilla Public
+// License, v. 2.0. If a copy of the MPL was not distributed with this
+// file, You can obtain one at https://mozilla.org/MPL/2.0/.
+
+//! STM32H7-specific USART details.
+
+pub use drv_stm32xx_sys_api::Sys;
+
+use super::BaudRate;
+
+#[cfg(feature = "stm32h743")]
+use stm32h7::stm32h743 as device;
+
+#[cfg(feature = "stm32h753")]
+use stm32h7::stm32h753 as device;
+
+/// Which USART device
+#[derive(Debug, Clone, Copy)]
+pub enum DeviceId {
+    Usart3,
+}
+
+pub struct Device(&'static device::usart1::RegisterBlock);
+
+impl Device {
+    /// Turn on the `device` USART with the given baud rate.
+    // TODO passing in a `Sys` seems weird. Should we take the sys task ID
```

Comment:
If you're IPCing, the `Sys` is the handle to the remote server, so, it's probably the right thing to pass in.